twingate/client_connector_tokens_test.go

```diff
@@ -72,23 +65,43 @@ func TestClientConnectorTokensVerifyError(t *testing.T) {
 	// response JSON
 	verifyTokensOkJson := `{}`
 
-	r := ioutil.NopCloser(bytes.NewReader([]byte(verifyTokensOkJson)))
-	client := createTestClient()
+	client := newHTTPMockClient()
+	defer httpmock.DeactivateAndReset()
 
 	accessToken := "test1"
 	refreshToken := "test2"
 
-	GetDoFunc = func(req *retryablehttp.Request) (*http.Response, error) {
-		header := req.Header.Get("Authorization")
-		assert.Contains(t, header, accessToken)
-		return &http.Response{
-			StatusCode: 501,
-			Body:       r,
-		}, nil
-	}
+	httpmock.RegisterResponder("POST", client.APIServerURL+"/access_node/refresh",
+		func(req *http.Request) (*http.Response, error) {
+			header := req.Header.Get("Authorization")
+			assert.Contains(t, header, accessToken)
+			return httpmock.NewStringResponse(501, verifyTokensOkJson), nil
+		})
+
 	err := client.verifyConnectorTokens(refreshToken, accessToken)
 
-	assert.EqualError(t, err, "failed to verify connector tokens: request  failed, status 501, body {}")
+	assert.EqualError(t, err, "failed to verify connector tokens with id : request  failed, status 501, body {}")
```

Comment:
The connector id is missing in the error.
Also based on the double space between the `request` and `failed` there is something missing there too.

Reply:
there should not be connectorID in error, it has to be just APIError (without id)

Reply:
```go
func NewAPIErrorWithID(wrappedError error, operation string, resource string, id graphql.ID) *APIError {
	return &APIError{
		WrappedError: wrappedError,
		Operation:    operation,
		Resource:     resource,
		ID:           id,
	}
}

func NewAPIError(wrappedError error, operation string, resource string) *APIError {
	return &APIError{
		WrappedError: wrappedError,
		Operation:    operation,
		Resource:     resource,
		ID:           "",
	}
}
```
i have mixed feelings about these 2 errors and why we need the one with ID ? what do you think @liorr

Reply:
I am OK with it as it is